packages/lodestar/src/sync/regular/naive/naive.ts

```diff
@@ -154,12 +171,41 @@ export class NaiveRegularSync implements IRegularSync {
   }
 
   private getSyncPeers = async (): Promise<PeerId[]> => {
-    return this.network.getPeers().reduce( (validPeers: PeerId[], peer: PeerId) => {
-      const rep = this.reps.getFromPeerId(peer);
-      if(rep && rep.latestStatus) {
-        validPeers.push(peer);
+    if (!this.network.getPeers().includes(this.bestPeer)) {
+      // bestPeer disconnected
+      this.bestPeer = undefined;
+      await this.waitForBestPeer();
+    }
+    return [this.bestPeer];
+  };
+
+  private waitForBestPeer = async (): Promise<void> => {
+    // check every slot
+    const waitingTime = this.config.params.SECONDS_PER_SLOT * 1000;
+    const state = await this.chain.getHeadState();
+    const status = createStatus(this.chain);
+    this.controller = new AbortController();
+    let isAborted = false;
+    this.controller.signal.addEventListener("abort", () => {
+      this.logger.verbose("RegularSync: Abort waitForBestPeer");
+      isAborted = true;
+    });
+    while (!this.bestPeer && !isAborted) {
+      const peers = this.network.getPeers();
+      const previousSlot = getCurrentSlot(this.config, state.genesisTime) - 1;
+      await syncPeersStatus(this.reps, this.network, status);
+      const maxHeadSlot = Math.max(...peers.map(
+        (peerId) => this.reps.get(peerId.toB58String()).latestStatus?.headSlot || 0));
+      this.bestPeer = peers.find(peerId => {
+        const headSlot = this.reps.get(peerId.toB58String()).latestStatus?.headSlot;
+        return headSlot >= previousSlot && headSlot === maxHeadSlot;
+      });
+      await sleep(waitingTime);
```

Comment:
should we be waiting _before_ we send statuses, so we get a more current status?